config/config.go

```diff
@@ -881,12 +887,12 @@ func TestBlockSyncConfig() *BlockSyncConfig {
 // ValidateBasic performs basic validation.
 func (cfg *BlockSyncConfig) ValidateBasic() error {
 	switch cfg.Version {
-	case "v0":
+	case v0:
 		return nil
-	case "v1", "v2":
-		return fmt.Errorf("blocksync version %s has been deprecated. Please use v0 instead", cfg.Version)
+	case v1, v2:
+		return ErrDeprecatedBlocksyncVersion{Version: cfg.Version, Allowed: AllowedBlockSyncConfigVersions}
```

Comment:
You are defining `AllowedBlockSyncConfigVersions` (for the error itself), but the _switch_ doesn't use it. So we will have to be careful in the future when modifying the blocksync versions to make sure that what shows in the _switch_ is consistent with the definition of `AllowedBlockSyncConfigVersions`.
My suggestion,

* either you get rid of `AllowedBlockSyncConfigVersions` and use `v0` directly hard-coded in line 893 (anyway it is hard-coded in the _case_ in line 890, isn't it?),
* or you modify the switch to use the information contained in `AllowedBlockSyncConfigVersions`

I'd personally lean toward the first (simpler) option.

Reply:
makes sense, thanks for advice! gonna lean towards the first one as well